### ISSUE TITLE
[Spike] Discuss incident management procedure

### ISSUE THREAD

**Issue by hemajv** [operate-first/SRE#14]
We should implement an incident management procedure to follow for handling outages/incidents related to the ODH services/applications on MOC

Previously we followed: https://gitlab.cee.redhat.com/data-hub/dh-runbooks/-/blob/master/incident_management_procedure.md and tracked incidents through JIRA/OnePortal. For starters, we could track our outages as issues in this repo?

cc @anishasthana @HumairAK @durandom

**Comment by durandom** [comment]
cc @goern @harshad16 
do you guys have anything that compares to PagerDuty for this?

I think tracking issues in GH (but another repo than this) makes sense. Although it's an abuse. But the APIs for exporting data and creating issues are well known.
There's also an alertmanager github receiver https://github.com/m-lab/alertmanager-github-receiver

**Comment by goern** [comment]
we have alertmanager on internal data hub...but no procedures documents

**Comment by hemajv** [comment]
some alternatives I found for PagerDuty (but not all are actively being maintained):

1. [Cabot](https://github.com/arachnys/cabot) - However, the project is not actively being maintained
2. [OpenDuty](https://github.com/openduty/openduty) - Currently in a beta version (don't see a lot of activity in the repo, might be discontinued)
3. [Dispatch](https://github.com/Netflix/dispatch) - Incident management tool by Netflix
4. [Response](https://github.com/monzo/response) - Django based incident tool

**Comment by anishasthana** [comment]
Is there a specific reason we'd not want to use PagerDuty? It works very well for what it needs to do. Internally we also use Jira tickets and One Portal to track incidents, so maybe using Github issues is not _that_ much of an abuse. It'll be easier for the community at large to also see how operate-first works through incidents and all that.

**Comment by HumairAK** [comment]
@hemajv  so based on your comments 1) and 2) don't seem like something we want to use due to lack of support. 

And looking into 3) Dispatch, it seems this requires slack and 4) seems it requires us to use jira for issue tracking (correct me if I'm wrong). So unless we're willing to switch to those platforms, those would also be a no-go. 

I am not against using github issues for now, we can make a new project board called `incidents` and keep the issues for incidents confined to this repo with appropriate labels etc.

**Comment by hemajv** [comment]
@HumairAK yep, most of the options I found primarily support slack and JIRA integrations for communication/tracking purposes. I did also find few other tools like [Squadcast](https://www.squadcast.com/pricing) which has a free plan option, however it comes with limitations on number of users, services etc.

Having said that, we can start by creating issues and tracking them in a new project board and as we onboard more services we could switch to something else?
